File: example/android/app/src/hms/java/com/exponea/example/MessageService.java
package com.exponea.example;

import android.app.NotificationManager;
import android.content.Context;
import androidx.annotation.NonNull;
import com.exponea.ExponeaModule;
import com.huawei.hms.push.HmsMessageService;
import com.huawei.hms.push.RemoteMessage;

public class MessageService extends HmsMessageService {

    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);
        ExponeaModule.Companion.handleRemoteMessage(
                getApplicationContext(),
                remoteMessage.getDataOfMap(),
                (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE));
    }

    @Override
    public void onNewToken(@NonNull String token) {
        super.onNewToken(token);
        ExponeaModule.Companion.handleNewHmsToken(
                getApplicationContext(),
                token);
    }
}

File: example/android/app/src/hms/java/exponea/example/MessageService.kt
package exponea.example

import android.app.NotificationManager
import android.content.Context
import com.exponea.ExponeaModule
import com.huawei.hms.push.HmsMessageService
import com.huawei.hms.push.RemoteMessage

class MessageService : HmsMessageService() {

    override fun onMessageReceived(message: RemoteMessage) {
        super.onMessageReceived(message)
        ExponeaModule.handleRemoteMessage(
            applicationContext,
            message.dataOfMap,
            getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
        )
    }

    override fun onNewToken(token: String) {
        super.onNewToken(token)
        ExponeaModule.handleNewHmsToken(applicationContext, token)
    }
}
